From dcfd7dbbf798fcd4c492584e36f2eb4cbaf5e6ad Mon Sep 17 00:00:00 2001
From: Jurgen Kramer <gtmkramer@xs4all.nl>
Date: Mon, 9 Nov 2015 13:16:53 +0100
Subject: [PATCH] usb: Add native DSD support for Hegel HD12 DSD

This patch adds native DSD support for the Hegel HD12 DSD DAC.
As this DAC is using a C-Media based USB solution I have adjusted the comment
regarding XMOS.

Signed-off-by: Jurgen Kramer <gtmkramer@xs4all.nl>
---
 sound/usb/quirks.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index 5ca80e7..d8c49a1 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -1262,7 +1262,10 @@ u64 snd_usb_interface_dsd_format_quirks(struct snd_usb_audio *chip,
 		}
 	}
 
-	/* XMOS based USB DACs */
+	/* USB DACs with native DSD support
+	 * Vendor 0x20b1: XMOS based
+	 * Vendor 0x0d8c: C-Media based
+	 */
 	switch (chip->usb_id) {
 	case USB_ID(0x20b1, 0x3008): /* iFi Audio micro/nano iDSD */
 	case USB_ID(0x20b1, 0x2008): /* Matrix Audio X-Sabre */
@@ -1271,6 +1274,7 @@ u64 snd_usb_interface_dsd_format_quirks(struct snd_usb_audio *chip,
 			return SNDRV_PCM_FMTBIT_DSD_U32_BE;
 		break;
 
+	case USB_ID(0x0d8c, 0x0316): /* Hegel HD12 DSD */
 	case USB_ID(0x20b1, 0x000a): /* Gustard DAC-X20U */
 	case USB_ID(0x20b1, 0x2009): /* DIYINHK DSD DXD 384kHz USB to I2S/DSD */
 	case USB_ID(0x20b1, 0x2023): /* JLsounds I2SoverUSB */
-- 
2.1.0

